### PR TITLE
Feature/inspector layout section toggle visibility

### PR DESCRIPTION
**Pull request by enidemi**
**Problem:**
The flex element layout section shows subsections with empty values.

**Fix:**
Hide subsections that have no values set, the subsections can be opened with the button. Without values the buttons have 30% opacity. Toggled buttons have primary colored background with white text.
Hide flex element layout section when all of the controls are not set.
Performance is not optimized for this PR as this is still experimental.

**Commit Details:**
- this valuebased off toggle state is only for flex layout 
- lots of smaller hooks to find the initial value for each subsection
- one hook that uses the smaller hooks to find the initial value for the layout section
- the automatic toggling based on metadata is done with usePropControlledStateV2

### CODE REVIEW

File: editor/src/components/inspector/sections/layout-section/flex-element-subsection/flex-element-subsection.tsx [hunk 1]
```diff
@@ -65,59 +72,96 @@ export const FlexElementSubsectionExperiment = betterReactMemo(
...
 
-    const isMainAxisVisible =
-      isMainAxisVisibleForFlexDirection ||
-      isNotUnsetDefaultOrDetected(alignSelf.controlStatus) ||
-      isNotUnsetDefaultOrDetected(flexBasis.controlStatus) ||
-      isNotUnsetDefaultOrDetected(flexGrow.controlStatus) ||
-      isNotUnsetDefaultOrDetected(flexShrink.controlStatus)
+export function useInitialAdvancedSectionState(): boolean {
+  const alignSelf = useInspectorLayoutInfo('alignSelf')
+  return isNotUnsetDefaultOrDetected(alignSelf.controlStatus)
```
Comment (Rheeseyb) [inline editor/src/components/inspector/sections/layout-section/flex-element-subsection/flex-element-subsection.tsx:96]: I really struggle to mentally parse this function name. It's effectively saying "the user has explicitly given this thing a value", right?

Comment (enidemi) [reply editor/src/components/inspector/sections/layout-section/flex-element-subsection/flex-element-subsection.tsx:96]: yes :D

Comment (Rheeseyb) [reply editor/src/components/inspector/sections/layout-section/flex-element-subsection/flex-element-subsection.tsx:96]: Ok I might rename this in the near future then :D

Comment (enidemi) [reply editor/src/components/inspector/sections/layout-section/flex-element-subsection/flex-element-subsection.tsx:96]: there are a scary amount of controlstatuses

File: editor/src/components/inspector/sections/layout-section/flex-element-subsection/flex-element-subsection.tsx [hunk 2]
```diff
@@ -65,59 +72,96 @@ export const FlexElementSubsectionExperiment = betterReactMemo(
...
-              collapsed={!mainAxisControlsOpen}
-              selected={false}
-            />
-          </SquareButton>
+          <div style={{ display: 'flex', justifyContent: 'flex-end' }}>
+            <Button highlight onClick={toggleFixedSection} primary={fixedControlsOpen}>
+              <InlineLink
+                style={{
+                  color: fixedControlsOpen ? colorTheme.white.value : colorTheme.primary.value,
+                  opacity: !initialIsFixedSectionVisible && !fixedControlsOpen ? 0.3 : 1,
```
Comment (Rheeseyb) [inline editor/src/components/inspector/sections/layout-section/flex-element-subsection/flex-element-subsection.tsx:141]: Can you flip the condition here please? `initialIsFixedSectionVisible || fixedControlsOpen ? 1 : 0.3`. And likewise below

File: editor/src/components/inspector/sections/layout-section/layout-section.tsx [hunk 3]
```diff
@@ -5,6 +5,7 @@ import { emptySpecialSizeMeasurements } from '../../../../core/shared/element-te
 import { betterReactMemo } from '../../../../uuiui-deps'
 import { useEditorState } from '../../../editor/store/store-hook'
 import { fastForEach } from '../../../../core/shared/utils'
+import * as EP from '../../../../core/shared/element-path'
```
Comment (Rheeseyb) [inline editor/src/components/inspector/sections/layout-section/layout-section.tsx:8]: No longer needed?

**Review by Rheeseyb — APPROVED** [review]

**Pull request merged**